kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/S3WriterManager.scala

```diff
@@ -215,12 +221,22 @@ object S3WriterManager {
     val formatWriterFn: (TopicPartition, Map[PartitionField, String]) => S3FormatWriter = (topicPartition, partitionValues) =>
       config.bucketOptions.find(_.sourceTopic == topicPartition.topic.value) match {
         case Some(bucketOptions) =>
-          val fileNamingStrategy = fileNamingStrategyFn(topicPartition.topic)
-
-          val path: BucketAndPath = fileNamingStrategy
-            .stagingFilename(bucketOptions.bucketAndPrefix, topicPartition, partitionValues)
-          val size: Int = minAllowedMultipartSizeFn()
-          S3FormatWriter(bucketOptions.formatSelection, outputStreamFn(path, size))
+          bucketOptions.writeMode match {
+            case S3WriteMode.Streamed => val fileNamingStrategy = fileNamingStrategyFn(topicPartition.topic)
```

Comment:
See previous comment. The write mode can also be responsible for handling this code. it would make it easier and more composable. Here you would have the writeMode and from there just call .

You have this code:

```
val path: BucketAndPath = fileNamingStrategy.stagingFilename(bucketOptions.bucketAndPrefix, topicPartition, partitionValues)
``` 
duplicated. 

Then 
```bucketOptions.localBuildDirectory``` should never be Option. If you have a Local write mode this should be set. This will simplify the code and won't require checks on an invalid state because wouldn't have been created in the first place.

Reply:
I've made a bunch of changes around the configuration so that it is much improved and more straightforward.